### PR TITLE
KAFKA-14575: Move ClusterTool to tools module

### PR DESCRIPTION
**Pull request by mimaison**
### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: tools/src/main/java/org/apache/kafka/tools/ClusterTool.java [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+                .help("Get information about the ID of a cluster.");
+        Subparser unregisterParser = subparsers.addParser("unregister")
+                .help("Unregister a broker.");
+        for (Subparser subpparser : Arrays.asList(clusterIdParser, unregisterParser)) {
+            subpparser.addArgument("--bootstrap-server", "-b")
+                    .action(store())
+                    .help("A list of host/port pairs to use for establishing the connection to the kafka cluster.");
+            subpparser.addArgument("--config", "-c")
+                    .action(store())
+                    .help("A property file containing configs to passed to AdminClient.");
```
Comment (fvaleri) [inline tools/src/main/java/org/apache/kafka/tools/ClusterTool.java]:
```suggestion
                    .help("A property file containing custom configs for the AdminClient.");
```

File: tools/src/main/java/org/apache/kafka/tools/ClusterTool.java [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+
+import static net.sourceforge.argparse4j.impl.Arguments.store;
+
+public class ClusterTool {
+
+    public static void main(String... args) {
+        Exit.exit(mainNoExit(args));
+    }
+
+    static int mainNoExit(String... args) {
```
Comment (fvaleri) [inline tools/src/main/java/org/apache/kafka/tools/ClusterTool.java:42]: Why not simply embed this logic inside the main method to have one less stack frame?

**Review by fvaleri — COMMENTED** [review]
I tried your changes on a running cluster and they work fine.
Left few comments.
Thanks.

File: tools/src/main/java/org/apache/kafka/tools/ClusterTool.java (hunk 1 above)
Comment (mimaison) [reply tools/src/main/java/org/apache/kafka/tools/ClusterTool.java]: Thanks, I've reworded that message

File: tools/src/main/java/org/apache/kafka/tools/ClusterTool.java (hunk 2 above)
Comment (mimaison) [reply tools/src/main/java/org/apache/kafka/tools/ClusterTool.java:42]: The idea was to make it like `MetadataQuorumCommand`. In the future we will be able to put this method in a Command interface or utils class.

Comment (fvaleri) [reply tools/src/main/java/org/apache/kafka/tools/ClusterTool.java:42]: I like the idea of the command interface. In that case, we should add the suffix `*Command` to all implementing classes (e.g. `ClusterToolCommand`).

Comment (mimaison) [reply tools/src/main/java/org/apache/kafka/tools/ClusterTool.java:42]: Yes. I was thinking of doing that after the Scala to Java conversion.

**Comment by fvaleri** [comment]
LGTM. Thanks.

**Review by dengziming — APPROVED** [review]
LGTM, I find that we didn't have a 1bin/windows/kafka-cluster.bat1 currently, how about added it here as a connivence.

**Comment by mimaison** [comment]
@dengziming I opened https://issues.apache.org/jira/browse/KAFKA-14614 to look at the missing script for Windows

**Review by dengziming — APPROVED** [review]
LGTM.

**Pull request merged**